draft-ietf-tls-wkech.xml

```diff
@@ -345,120 +342,147 @@ Client <-->|                    | |                    | facing
-                        value octet is stored as a single Unicode codepoint <xref
-                        target="ISOMORPHIC-DECODE"/>.  In almost all cases, this is
-                        equivalent to the ordinary string representation of the presentation
-                        value.
-                    </t>
-                    <t>
-                        For list-valued SvcParams (e.g., "alpn"), the value is a JSON Array
-                        of Strings.  Each String represents an octet sequence, as in the
-                        single-value case.
-                    </t>
-                </list>
-            </t>
-        </list>
-        An empty endpoint object corresponds to an HTTPS record with inferred
-        SvcPriority, TargetName=".", and no ECH support.  An empty record of
-        this kind can be useful as a simple way to make use of the HTTPS RR
+	with an "endpoints" key that contains an array of objects.
+        All other keys MUST be ignored.
+        The length of the list implies restrictions and meaning to the content
+        of the entries:
+    <list style="symbols">
+      <t>
+        An empty endpoints array corresponds to an HTTPS record with inferred
+        SvcPriority, the TargetName equal to ".", and no ECH support.  This can
+        can be useful as a simple way to make use of the HTTPS RR
         type's HSTS behavior.
+      </t>
+      <t>
+	An endpoints array with one element can be either a ServiceMode
+	entry, containing at least one key from the JSON HTTP Origin Info
+	registry (see <xref target="iana">IANA Considerations</xref>, below)
+	or an AliasMode entry that points to another DNS name that must
+        be resolved.
+      </t>
+      <t>
+	If the endpoints array has more than one element, every item MUST
+        be a ServiceMode entry.
+      </t>
+    </list>
     </t>
-
     <t>
-        If the ZF is unable to convert the JSON into a DNS zone (e.g., due to an
-        unrecognized SvcParamKey), or if the resulting zone fails validation checks,
-        the ZF MUST NOT update the DNS. Such failures will not be directly visible to
-        the client-facing server, so ZF implementations will need to provide some
-        form of reporting so that the situation can be resolved.
-        Note that this can lead to inconsistent behavior for a single origin served by multiple ZFs.
+	The intent of these restrictions is to reduce the chance of misconfiguration
+	by the ZF and to avoid introducing semantics not defined by
+        <xref target="RFC9460"/>.
     </t>
 
-    <t>
-        A ZF MAY apply additional processing according to its own policy, such as
-        adjusting TTL values and correcting common misconfigurations.
+    <t>The following keys are defined for both types of entries:
+    <list style="symbols">
+      <t>
+	  regeninterval: the number of seconds between key generation
+	  actions at the origin, i.e. a replacement ECHConfigList may be
+	  generated this often.
```

Comment:
`regeninterval` is an attribute of the outer object, not the objects in the array.

Reply:
That's not what the samples show.

Reply:
OK, let's fix that.  The top-level object should have two keys: "endpoints" and "regeninterval".

Reply:
Making regeninterval top-level fields have a two-part fix: here and in #29